nimare/io.py

```diff
@@ -275,3 +278,123 @@ def convert_sleuth_to_dataset(text_file, target="ale_2mm"):
+    collection_ids, contrasts, img_dir=None, map_type_conversion=None, **dset_kwargs
+):
+    """
+    Convert a group neurovault collections into a NiMARE Dataset.
+
+    Parameters
+    ----------
+    collection_ids : :obj:`list` of :obj:`int` or :obj:`dict`
+        A list of collections on neurovault specified by their id.
+        The collection ids can accessed through the neurovault API
+        (i.e., https://neurovault.org/api/collections) or
+        their main website (i.e., https://neurovault.org/collections).
+        For example, in this URL https://neurovault.org/collections/8836/,
+        `8836` is the collection id.
+        collection_ids can also be a dictionary whose keys are the informative
+        study name and the values are collection ids to give the collections
+        more informative names in the dataset.
+    contrasts : :obj:`dict`
+        Dictionary whose keys represent the name of the contrast in
+        the dataset and whose values represent how that contrast is identified
+        in neurovault.
+        For example, under the ``Name`` column in this URL
+        https://neurovault.org/collections/8836/,
+        a valid contrast could be "as-Animal", which will be called "animal" in the created
+        dataset if the contrasts argument is ``{'animal': ["as-Animal"]}``.
+    img_dir : :obj:`str` or None
+        Base path to save all the downloaded images, by default the images
+        will be saved to a temporary directory with the prefix "neurovault"
+    map_type_conversion : :obj:`dict` or None
+        Dictionary whose keys are what you expect the `map_type` name to
+        be in neurovault and the values are the name of the respective
+        statistic map in a nimare dataset. Default = None.
+    **dset_kwargs : keyword arguments passed to Dataset
+        Keyword arguments to pass in when creating the Dataset object.
+        see :obj:`nimare.dataset.Dataset` for details.
+
+    Returns
+    -------
+    :obj:`nimare.dataset.Dataset`
+        Dataset object containing experiment information from neurovault.
+    """
+    if img_dir is None:
+        img_dir = Path(tempfile.mkdtemp(prefix="neurovault_"))
+    else:
+        img_dir = Path(img_dir, exist_ok=True)
+        img_dir.mkdir(parents=True, exist_ok=True)
+
+    if map_type_conversion is None:
+        map_type_conversion = {
+            "T map": "t",
+            "variance": "varcope",
+            "univariate-beta map": "beta",
+        }
+
+    if not isinstance(collection_ids, dict):
+        collection_ids = {nv_coll: nv_coll for nv_coll in collection_ids}
+
+    dataset_dict = {}
+    for coll_name, nv_coll in collection_ids.items():
+        for contrast_name, contrast_ids in contrasts.items():
+            nv_url = f"https://neurovault.org/api/collections/{nv_coll}/images/?format=json"
+            images = requests.get(nv_url).json()
+
+            dataset_dict[f"study-{coll_name}"] = {
+                "contrasts": {
```

Comment:
See above comment—I think the default record for the study should be created outside the loop over contrasts, as it only needs to be done once. (Plus I think if there are multiple contrasts, this will overwrite all but the last. Might be worth adding a test that has multiple contrasts.)

Reply:
d'ohx2, fixed this too.